android/filament-android/src/main/java/com/google/android/filament/TransformManager.java

```diff
@@ -199,6 +200,46 @@ public int getParent(@EntityInstance int i) {
         return nGetParent(mNativeObject, i);
     }
 
+
+    /**
+     * Returns the number of children of an {@link EntityInstance}.
+     *
+     * @param i the {@link EntityInstance} of the transform component to query.
+     * @return The number of children of the queried component.
+     */
+    public int getChildCount(@EntityInstance int i) {
+        return nGetChildCount(mNativeObject, i);
+    }
+
+    /**
+     * Gets a list of children for a transform component.
+     *
+     * @param i the {@link EntityInstance} of the transform component to get the children from.
+     * @param count The maximum number of children to retrieve.
+     * @return Array of retrieved children {@link Entity}.
+     */
+    @Entity
+    @NonNull
+    public int[] getChildren(@EntityInstance int i, @IntRange(from = 1) int count) {
+        int[] entities = new int[count];
+        getChildren(i, entities);
+        return entities;
+    }
+
+    /**
+     * Gets a list of children for a transform component.
+     *
+     * @param i the {@link EntityInstance} of the transform component to get the children from.
+     * @param outEntities array to receive the result sized to the maximum number of children to
+     *                    retrieve
+     */
+    @Entity
+    @NonNull
+    public void getChildren(@EntityInstance int i, @Size(min = 1) int[] outEntities) {
+        if (outEntities.length < 1) throw new ArrayIndexOutOfBoundsException("outEntities must be at least 1");
+        nGetChildren(mNativeObject, i, outEntities, outEntities.length);
+    }
+
```

Comment:
I think these API should follow the pattern we've used elsewhere. That is, you only have the version that takes `outEntitites` and if it is null then you allocate it.

Reply:
```suggestion
    /**
     * Gets a list of children for a transform component.
     *
     * @param i the {@link EntityInstance} of the transform component to get the children from.
     * @return Array of retrieved children {@link Entity}.
     */
    public @Entity @NonNull int[] getChildren(@EntityInstance int i) {
        return getChildren(i, null);
    }

    /**
     * Gets a list of children for a transform component.
     *
     * @param i        the {@link EntityInstance} of the transform component to get the children
     *                 from.
     * @param entities array to receive the result sized to the maximum number of children to
     *                 retrieve.
     */
    public @Entity @NonNull int[] getChildren(@EntityInstance int i, @Nullable @Size(min = 1) int[] entities) {
        if (entities == null) {
            entities = new int[getChildCount(i)];
        }
        if (entities.length > 0) {
            nGetChildren(mNativeObject, i, entities, entities.length);
        }
        return entities;
    }

```